integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java

```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+        if (references.isPresent() && previousProcessorState != null) {
+            //reset the processor state
+            Map<String, RevisionDTO> finalComponentRevisions = processorRevisions;
+
+            //if all referencing processors of of the same state we can call the quick rest endpoint to update all of them
+            Set<String> distinctStates = previousProcessorState.values().stream().collect(Collectors.toSet());
+            if (distinctStates.size() > 0) {
+                if (distinctStates.size() == 1) {
+                    String state = new ArrayList<>(distinctStates).get(0);
+                    if(!NifiProcessUtil.PROCESS_STATE.STOPPED.name().equals(state)) {
+                        UpdateControllerServiceReferenceRequestEntity startComponentsRequest = new UpdateControllerServiceReferenceRequestEntity();
+
+                        startComponentsRequest.setState(state);
+                        startComponentsRequest.setReferencingComponentRevisions(finalComponentRevisions);
+                        startComponentsRequest.setId(controllerService.getId());
+                        updateReferences(id, startComponentsRequest);
+                        log.info("Updating all component references to be {} for controller service  {} ", state, controllerService.getName());
+                        boolean updatedReferences = ensureComponentsAreOfState(id, "Processor", state, 5, 500, TimeUnit.MILLISECONDS);
+                        if (!updatedReferences) {
+                            //error unable to change the state of the references. ... error
+                            throw new NifiClientRuntimeException("The controller service {} was updated, but it was unable to update the state of the processors as " + state
+                                                                 + ".  Please visit NiFi to reconcile and fix any controller service issues. " + controllerService.getName());
+                        }
+                        log.info("Successfully updated controller service {}", controllerService.getName());
+                    }
+                } else {
+                    log.info(
+                        "The controller service component references ({} total) had mixed states prior to updating {}.  Going through each processor/component and setting its state back to what it was prior to the update.",
+                        previousProcessorState.size(), distinctStates);
+                    //update references back to previous states
+                    List<ProcessorEntity> updatedProcessors = references.get().getControllerServiceReferencingComponents().stream()
+                        .filter(c -> c.getComponent().getReferenceType().equalsIgnoreCase("PROCESSOR"))
+                        .filter(c-> !c.getComponent().getState().equals(NifiProcessUtil.PROCESS_STATE.STOPPED.name()))
+                        .map(referencingComponentEntity -> referencingComponentEntity.getComponent()).map(c -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)